pandas/sparse/scipy_sparse.py

```diff
@@ -0,0 +1,102 @@
+"""
+Interaction with scipy.sparse matrices.
+
+Currently only includes SparseSeries.to_coo helpers.
+"""
+from pandas.core.frame import DataFrame
+from pandas.core.index import MultiIndex, Index
+from pandas.core.series import Series
+import itertools
+import numpy
+from pandas.compat import OrderedDict
+from pandas.tools.util import cartesian_product
+
+
+def _get_label_to_i_dict(labels, sort_labels=False):
+    """ Return OrderedDict of unique labels to number. Optionally sort by label. """
+    labels = Index(map(tuple, labels)).unique().tolist()  # squish
+    if sort_labels:
+        labels = sorted(list(labels))
+    d = OrderedDict((k, i) for i, k in enumerate(labels))
+    return(d)
+
+
+def _get_index_subset_to_coord_dict(index, subset, sort_labels=False):
+    ilabels = list(zip(*[index.get_level_values(i) for i in subset]))
```

Comment:
to do this in a vectorized way, I think this might suffice:
```
new_index = pd.MultiIndex([index.levels[i] for i in subset],
                          [index.labels[i] for i in subset])
```